openmls/src/treesync/treesync_node.rs

```diff
@@ -59,7 +62,10 @@ impl TreeSyncNode {
     pub(in crate::treesync) fn node_without_private_key(&self) -> Option<Node> {
         if let Some(node) = self.node() {
             match node {
-                Node::LeafNode(leaf_node) => Node::LeafNode(leaf_node.key_package().clone().into()),
+                Node::LeafNode(leaf_node) => Node::LeafNode(LeafNode::new_with_ref(
+                    leaf_node.key_package().clone(),
+                    leaf_node.key_package_ref().cloned(),
```

Comment:
Would it make sense to just implement `Copy` for `HashRef` and leave optimization to the compiler?

Reply:
It implements `Copy`. But this is an `Option<&_>` that we have to convert, hence the `cloned`.

Reply:
Ah, I hadn't looked close enough. Fair point.